editoast/src/error.rs

```diff
@@ -0,0 +1,114 @@
+use actix_web::{error::JsonPayloadError, http::StatusCode, HttpResponse, ResponseError};
+use diesel::result::Error as DieselError;
+use redis::RedisError;
+use serde::Serialize;
+use serde_json::{json, Map, Value};
+use std::result::Result as StdResult;
+use std::{
+    error::Error,
+    fmt::{Display, Formatter},
+};
+
+pub type Result<T> = StdResult<T, InternalError>;
+
+/// Trait for all errors that can be returned by editoast
+pub trait EditoastError: Error + Send + Sync {
+    fn get_status(&self) -> StatusCode;
+
+    fn get_type(&self) -> &'static str;
+
+    fn extra(&self) -> Map<String, Value> {
+        Default::default()
+    }
+}
+
+#[derive(Debug, Serialize)]
+pub struct InternalError {
+    #[serde(skip)]
+    status: StatusCode,
+    error_type: &'static str,
+    #[serde(flatten)]
+    exta: Map<String, Value>,
+    message: String,
+}
```

Comment:
s/exta/extra/
Also, #2697 defines the following format:

```
 "error": {
        "type": "no_path_found",
        "message": "No path could be found",
        "context": {}
    }
```

Shouldn't we start following that here? (Or should we update that format to the one used here - both options are fine by me)

Reply:
Done. I did match the format we defined at #2697